angular-ui/src/app/user-profile/user-profile.component.html

```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE html>
+<div>
+  <h1>{{ user.name }}</h1>
+  <p>{{ user.email }}</p>
+  <button (click)="editProfile()">Edit Profile</button>
```

Comment:
I actually think a link would be better. If so : <a (click)="editProfile()" routerLink="/edit-profile">Edit Profile</a>